server/controllers/calendarController.ts

```diff
@@ -28,12 +29,19 @@ export default class CalendarController {
         isMfa ? userAuthenticationService.getUserAuthenticationDetails(username) : [],
       ])
 
+      const alreadyDismissedExisting = alreadyDismissed(req, 'EXISTING_USER')
+      const alreadyDismissedNew = alreadyDismissed(req, 'NEW_USER')
+
       let mfaBanner = ''
       if (isMfa) {
         if (userAuthenticationDetails && userAuthenticationDetails.length > 0) {
-          mfaBanner = 'EXISTING_USER'
+          if (!alreadyDismissedExisting) {
```

Comment:
is the triple nesting of `if` necessary - could it be added onto the second level `if` instead?
Not a great fan of a mutable variable here either - difficult to tell what it will be set to under what circumstances - maybe better to split out to separate function that returns the `mfaBanner`?  What does the field mean - could it be an enum instead?

Reply:
would also be good to add some unit testing of this stuff

Reply:
Done some refactoring and added tests.